### PR TITLE
Implement grayscale support

### LINKED ISSUE #6: CI issue with docstring of UC8151D
There seems to be an issue with Sphinx parsing that docstring.  Currently being fixed by #4, but wanted to track just in case.

### PR DESCRIPTION
**Pull request by HDR**
Adds grayscale support for displays like GDEW0154T8D, based on https://github.com/adafruit/Adafruit_CircuitPython_IL0373/blob/main/adafruit_il0373.py

image

Test image used (github doesn't allow .bmp):
Grayscale

### CODE REVIEW

**Comment by tekktrik** [comment]
Hi @HDR!  Thanks for the submission!  It looks like the Github Actions CI had an issue with some of the formatting.  You can use `pre-commit` locally to auto-format everything and then submit a commit with the changes.  That will fix the CI error!

If you aren't familiar with using `pre-commit`, here's a tutorial that should help you get setup with it: https://learn.adafruit.com/creating-and-sharing-a-circuitpython-library/check-your-code

**Comment by HDR** [comment]
Looks like Sphinx is throwing an error now instead

File: adafruit_uc8151d.py
Comment (tekktrik) [inline adafruit_uc8151d.py]: I think it's getting confused by indentation like this, so it's not even your PR.  We updated the CI recently, so this is probably new.  I'm made a PR for it, and either you can add/merge in those changes and I can approve if it or you can wait for it to get merged if you'd like :)

**Review by tekktrik — COMMENTED** [review]
Looks like this is the problem:

**Comment by HDR** [comment]
> Looks like this is the problem:

Yep, already resolved it

**Comment by tekktrik** [comment]
2fast4me 😆

**Comment by tekktrik** [comment]
Unfortunately I don't have the hardware to test, but just wanted to help with everything preceding!

**Comment by ladyada** [comment]
nice work @HDR and thank you @tekktrik for helping shephard
@HDR can you add an example like the one photo'd so folks can try it?

**Comment by HDR** [comment]
> nice work @HDR and thank you @tekktrik for helping shephard @HDR can you add an example like the one photo'd so folks can try it?

Sure, i currently have this hooked up to Raspberry Pi Pico, example.zip contains a bmp and the example code.

```
import time
import board
import displayio
import busio
import adafruit_uc8151d

displayio.release_displays()

clk = board.GP10
si = board.GP11
dc = board.GP8
cs = board.GP9
rst = board.GP12
busy = board.GP13

display_bus = displayio.FourWire(
    busio.SPI(clk, si), command=dc, chip_select=cs, reset=rst, baudrate=1000000
)

time.sleep(1)

display = adafruit_uc8151d.UC8151D(
    display_bus, width=152, height=152, busy_pin=busy, rotation=180, grayscale=True
)


pic = displayio.OnDiskBitmap('Grayscale.bmp')
g = displayio.Group()
t = displayio.TileGrid(pic, pixel_shader=pic.pixel_shader)
g.append(t)
display.show(g)
display.refresh()
```


[Example.zip](https://github.com/adafruit/Adafruit_CircuitPython_UC8151D/files/8747805/Example.zip)

**Comment by ladyada** [comment]
instead of a bmp, could you rework the example to draw 4 rectangles/bitmap tiles that have a similar pattern? that way we dont have a file dependancy

**Comment by HDR** [comment]
> instead of a bmp, could you rework the example to draw 4 rectangles/bitmap tiles that have a similar pattern? that way we dont have a file dependancy


Here is a quick and simple example that does not depend on a bmp :)

```
import time
import board
import displayio
import busio
import adafruit_uc8151d

displayio.release_displays()

clk = board.GP10
si = board.GP11
dc = board.GP8
cs = board.GP9
rst = board.GP12
busy = board.GP13

display_bus = displayio.FourWire(
    busio.SPI(clk, si), command=dc, chip_select=cs, reset=rst, baudrate=1000000
)

time.sleep(1)

display = adafruit_uc8151d.UC8151D(
    display_bus, width=152, height=152, busy_pin=busy, rotation=180, grayscale=True
)


bitmap = displayio.Bitmap(152, 152, 4)

# Draw Black
for x in range(0,152):
    for y in range(0,38):
        bitmap[x, y] = 0
#Draw Dark Gray
for x in range(0,152):
    for y in range(38,76):
        bitmap[x, y] = 1
#Draw Light Gray
for x in range(0,152):
    for y in range(76,114):
        bitmap[x, y] = 2
#Draw White
for x in range(0,152):
    for y in range(114,152):
        bitmap[x, y] = 3

palette = displayio.Palette(4)
palette[0] = 0x000000 # Black
palette[1] = 0x404040 # Dark Gray
palette[2] = 0x808080 # Light Gray
palette[3] = 0xFFFFFF # White

g = displayio.Group()
t = displayio.TileGrid(bitmap, pixel_shader=palette)
g.append(t)
display.show(g)
display.refresh()

```

**Comment by ladyada** [comment]
perfect, can you add that to the PR as a new file? then we'll merge it together

**Comment by HDR** [comment]
Hopefully the Copyright and License stuff is correct, if not please let me know :)

**Review by ladyada — APPROVED** [review]

**Comment by ladyada** [comment]
ya got it, nice work!!!!

**Pull request merged**
